distributed/comm/tcp.py

```diff
@@ -414,11 +414,40 @@ def _check_encryption(self, address, connection_args):
             )
 
 
+async def _getaddrinfo(host, port, *, family, type=socket.SOCK_STREAM):
+    # If host and port are numeric, then getaddrinfo doesn't block and we can
+    # skip the whole thread thing, which seems worthwhile. So we try first
```

Comment:
From this code alone, it's not apparent what "the whole thread thing" is. I think it's worth mentioning that it is CPython that is dispatching to a thread in this case.